app/products/playbooks/screens/playbooks_runs/playbook_runs.tsx

```diff
@@ -0,0 +1,181 @@
+
+    const exit = useCallback(() => {
+        popTopScreen(componentId);
+    }, [componentId]);
+
+    useAndroidHardwareBackHandler(componentId, exit);
+
+    const [inProgressRuns, finishedRuns] = useMemo(() => {
+        const inProgress: PlaybookRunModel[] = [];
+        const finished: PlaybookRunModel[] = [];
+
+        allRuns.forEach((run) => {
+            if (run.endAt) {
+                finished.push(run);
+            } else {
+                inProgress.push(run);
+            }
+        });
+
+        return [inProgress, finished] as const;
+    }, [allRuns]);
+    const hasMoreFinishedRuns = useRef(true);
+    const finishedRunsPage = useRef(0);
+    const fetching = useRef(false);
+
+    const initialTab: TabsNames = inProgressRuns.length ? 'in-progress' : 'finished';
+    const [activeTab, tabsProps] = useTabs<TabsNames>(initialTab, tabs);
+
+    let data: Array<PlaybookRunModel | PlaybookRun> = inProgressRuns;
+    if (activeTab === 'finished') {
+        if (fetchedFinishedRuns.length) {
+            data = fetchedFinishedRuns;
+        } else {
+            data = finishedRuns;
+        }
+    }
+
+    const isEmpty = data.length === 0;
+
+    const renderItem: ListRenderItem<PlaybookRunModel> = useCallback(({item}) => {
+        return (
+            <PlaybookCard
+                run={item}
+                location={Screens.PLAYBOOKS_RUNS}
+            />
+        );
+    }, []);
+
+    const fetchFinishedRuns = useCallback(async () => {
+        if (fetching.current) {
+            return;
+        }
+        fetching.current = true;
+        const {runs, has_more, error} = await fetchFinishedRunsForChannel(serverUrl, channelId, finishedRunsPage.current);
+        fetching.current = false;
+        if (error) {
+            hasMoreFinishedRuns.current = false;
+            return;
+        }
+        hasMoreFinishedRuns.current = has_more ?? false;
+        finishedRunsPage.current++;
+        if (runs?.length) {
+            setFetchedFinishedRuns(runs);
+        }
+    }, [channelId, serverUrl]);
+
+    const onFinishedRunsReachEnd = useCallback(() => {
+        if (hasMoreFinishedRuns.current) {
+            fetchFinishedRuns();
+        }
+    }, [fetchFinishedRuns]);
+
+    useEffect(() => {
+        if (activeTab === 'finished' && hasMoreFinishedRuns.current && !finishedRuns.length) {
+            fetchFinishedRuns();
+        }
+    }, [activeTab, fetchFinishedRuns, finishedRuns.length]);
+
+    let content = (<EmptyState tab={activeTab}/>);
+    if (!isEmpty) {
+        content = (
+            <FlashList
+                data={data}
+                renderItem={renderItem}
+                contentContainerStyle={styles.container}
+                ItemSeparatorComponent={ItemSeparator}
+                estimatedItemSize={CARD_HEIGHT}
+                onEndReached={activeTab === 'finished' ? onFinishedRunsReachEnd : undefined}
+            />
+        );
+    }
```

Comment:
should content be memoized?

Reply:
Content is added directly into the JSX, not into a prop, so it is fine if it is not memoized.